Source/Foundation/Association.swift

```diff
@@ -50,10 +50,12 @@ public func associatedProperty<T: AnyObject>(host: AnyObject, keyPath: StaticStr
 /// This can be used as an alternative to `DynamicProperty` for creating strongly typed
 /// bindings on Cocoa objects.
 @warn_unused_result(message="Did you forget to use the property?")
-public func associatedProperty<Host: AnyObject, T>(host: Host, key: UnsafePointer<()>, @noescape initial: Host -> T, setter: (Host, T) -> ()) -> MutableProperty<T> {
+public func associatedProperty<Host: AnyObject, T>(host: Host, key: UnsafePointer<()>, @noescape initial: Host -> T, setter: (Host, T) -> (), setUp: (MutableProperty<T> -> ())? = nil) -> MutableProperty<T> {
     return associatedObject(host, key: key) { host in
         let property = MutableProperty(initial(host))
 
+        setUp?(property)
```

Comment:
I would prefer with it and and default { _ in }. This would be easier for the consumer, since he wouldn't have to reference self in the closure. There also so minor optimizations with it.